src/wallet/spend.cpp

```diff
@@ -745,6 +745,13 @@ static BResult<CreatedTransactionResult> CreateTransactionInternal(
     coin_selection_params.m_change_fee = coin_selection_params.m_effective_feerate.GetFee(coin_selection_params.change_output_size);
     coin_selection_params.m_cost_of_change = coin_selection_params.m_discard_feerate.GetFee(coin_selection_params.change_spend_size) + coin_selection_params.m_change_fee;
 
+    // The smallest change amount should be:
+    // 1. at least equal to dust threshold
+    // 2. at least 1 sat greater than fees to spend it at m_discard_feerate
+    const auto dust = GetDustThreshold(change_prototype_txout, coin_selection_params.m_discard_feerate);
+    const auto change_spend_fee = coin_selection_params.m_discard_feerate.GetFee(coin_selection_params.change_spend_size);
+    coin_selection_params.min_change = std::max(change_spend_fee + 1, dust);
```

Comment:
I'm not sure I understand the question about the remainder. I've tried to keep the existing behaviour unchanged.

We can totally add +1 to the dust threshold but that would be a change in behaviour. Are we sure we want to do it?

Reply:
The changeless window is based on `cost_of_change`, not just `change_spend_fee`. So the minimum viable change should be `cost_of_change + 1`.

I agree that the `+ 1` should not be applied to the dust threshold. The existing behavior throws away change when it is `< dust` or `<= cost of change`. So that's equivalent to `< dust` or `< cost of change + 1`.

Reply:
Err, yes. You're right, the `+1` should be on `coin_selection_params.m_cost_of_change`

Reply:
No. `cost_of_change` = `change_fee` + `change_spend_fee`. When we compare `change` To `min_viable_change` the `change_fee` is already deducted.

Think about it from this perspective, you don't want to create a change output that is less than what it takes to spend it.

Reply:
You're right thought. That that right now we drop change smaller than `cost_of_change` after deducting the `change_fee`. So the `change_fee` is deducted twice. Which I think is a bug. But I need to disclose this in the description.

Reply:
Ah ok. That makes sense.

Reply:
Okay, convinced that all sounds right. Thanks for explaining